File: togetherApp/ios/TogetherApi.m
//
//  TogetherApi.m
//  togetherApp
//
//  Created by carlos-valarezo-loaiza on 2/19/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TogetherApi, NSObject)
RCT_EXTERN_METHOD(registerUser: (NSDictionary *)data callback:(RCTResponseSenderBlock)callback)
@end

File: togetherApp/ios/TogetherApi.swift
//
//  TogetherApi.swift
//  togetherApp
//
//  Created by carlos-valarezo-loaiza on 2/19/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@objc(TogetherApi)
class TogetherApi: NSObject {
  
  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  func greetPeople(_ person: String) -> String{
    let greeting = "Hello " + person + " from swift..."
    print(greeting)
    return greeting
  }
  
  @objc func registerUser(_ data: NSDictionary, callback callback: RCTResponseSenderBlock){
    print(data)
    callback([data])
  }
}

